ledger/cryptarchia-ledger/src/coin.rs

```diff
@@ -0,0 +1,157 @@
+use crate::{crypto::Blake2b, Commitment, EpochState, LeaderProof, Nonce, Nullifier};
+use blake2::digest::Digest;
+use cryptarchia_engine::config::Config;
+use cryptarchia_engine::Slot;
+
+type SecretKey = [u8; 32];
+type PublicKey = [u8; 32];
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Clone, Debug, Eq, PartialEq, Copy, Hash)]
+pub struct Value(u32);
+
+impl From<u32> for Value {
+    fn from(value: u32) -> Self {
+        Self(value)
+    }
+}
+
+// This implementatio is only a stub
+// see https://github.com/logos-co/nomos-specs/blob/master/cryptarchia/cryptarchia.py for a spec
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+pub struct Coin {
+    sk: SecretKey,
+    nonce: Nonce,
+    value: Value,
+}
+
+impl Coin {
+    pub fn new(sk: SecretKey, nonce: Nonce, value: Value) -> Self {
+        Self { sk, nonce, value }
+    }
+
+    pub fn evolve(&self) -> Self {
+        let mut h = Blake2b::new();
+        h.update("coin-evolve");
+        h.update(self.sk);
+        h.update(self.nonce);
+        Self {
+            sk: self.sk,
+            nonce: <[u8; 32]>::from(h.finalize()).into(),
+            value: self.value,
+        }
+    }
+
+    fn value_bytes(&self) -> [u8; 32] {
+        let mut value_bytes = [0; 32];
+        value_bytes[28..].copy_from_slice(&self.value.0.to_be_bytes());
+        value_bytes
+    }
+
+    pub fn pk(&self) -> &PublicKey {
+        &self.sk
+    }
+
+    pub fn commitment(&self) -> Commitment {
+        let mut h = Blake2b::new();
+        h.update("coin-commitment");
+        h.update(self.nonce);
+        h.update(self.pk());
+        h.update(self.value_bytes());
+        <[u8; 32]>::from(h.finalize()).into()
+    }
+
+    pub fn nullifier(&self) -> Nullifier {
+        let mut h = Blake2b::new();
+        h.update("coin-nullifier");
+        h.update(self.nonce);
+        h.update(self.pk());
+        h.update(self.value_bytes());
+        <[u8; 32]>::from(h.finalize()).into()
+    }
```

Comment:
This is just mock right? As we do not have an specific cryptography scheme for this?